### PR TITLE
add other fixes on routes endpoint

### PR DESCRIPTION
**Pull request by UmuhireJessie**
### What does this PR do?
- Added how to check permissions if the URL has query parameters
- Changed a `getJourney()` function to have `.findOne()` instead of `.findAll()`
- Changed the `if` statement that checks if one of the `req.body` attributes is empty
- Changed `findAllPermissions()` function to use paginations which are query parameters 
- Added other examples in .env.example
- Fixes issues in resetting password by the user
- Made changes on `sendEmail()` function and its all usage
- Changed the `responseHandler()` function to adhere to the changes made before
- 
### Description of Task to be completed?
- Translation of all responses in the App
- Changing the design of the message (button hover effect) send to the client after registration, resetting and assigning 
- Making all controllers receive pure validated requests
### How should this be manually tested?
- After cloning the repo run `npm i` to install all packages
- For Windows users refer an extra step for installing [`redis`](https://www.youtube.com/watch?v=188Fy-oCw4w&ab_channel=ProgrammingKnowledge2)
- use postman to test different endpoints present in [`index.js`](https://github.com/atlp-rwanda/Dominators-Phantom-BE/blob/develop/src/routes/index.js)
- Run `npm run migrate` to migrate all models in the local database
- Run `npm run seed` to populate the tables with seeders' values
- Run `npm run dev` to start the server manually
- use `http://localhost:3000/api/v1/login` to log in by using admin credentials present in the seeder of the user table. (Change the port **3000** to adhere to your chosen port)
- After login and adding Authorisation in headers, insert the URL with the port and the method you want to use as prescribed in other PRs

### Any background context you want to provide?
The frontend had issues with fetching before which required us to modify the backend for it to work fine.
### Questions:

### CODE REVIEW

**Review by fab-ryan — APPROVED** [review]

File: .env.example [hunk 1]
```diff
@@ -1,17 +1,25 @@
...
 JWT_EXPIRES_IN=
 admintoken= 
 wrongtoken=
 USER_EMAIL= 
 USER_EMAIL_P= 
+PHANTOM_PASSWORD=
+APP_MAIL= 
+APP_MAIL_PASSWORD = 
+USER_LOCALHOST_PASSWORD= 
+USER_HOSTED_PASSWORD=
```
Comment (Anguandia) [inline .env.example:17]: The existing email-related variables are sufficient. Introducing more variables will only breed confusion. Please remove these variables. Individuals will only give appropriate values in their dev environments while the hosted environments will also be given corresponding appropriate values

File: src/controllers/passwordController.js [hunk 2]
```diff
@@ -3,42 +3,49 @@ import { resetLink } from '../helpers/sendEmail';
...
+  await userModel
+    .findOne({
+      where: { email: user_email },
     })
-}
-
-
+    .then((num) => {
+      null != num
+        ? (resetLink(user_email), msg(res, 200, 'Password reset link was sent'))
```
Comment (Anguandia) [inline src/controllers/passwordController.js]: Please translate this response too and other succeeding untranslated responses

File: src/controllers/passwordController.js [hunk 3]
```diff
@@ -3,42 +3,49 @@ import { resetLink } from '../helpers/sendEmail';
...
-    }
-
-    if (password != confirm)
-        return msg(res, 400, "Bad request, Password Mismatch")
-    const new_password = await bcrypt.hash(password, 12);
-    await userModel.update({ password: new_password }, {
-        where: { email }
-    }).then(num => {
-        1 == num ? msg(res, 200, req.t("updated_ok")) : msg(res, 401, req.t("updated_invalid_req"));
+  console.log('REQUEST', req.body);
```
Comment (Anguandia) [inline src/controllers/passwordController.js:28]: Remove this, and subsequent console.log statements

File: src/controllers/passwordController.js [hunk 4]
```diff
@@ -3,42 +3,49 @@ import { resetLink } from '../helpers/sendEmail';
...
+  const confirm = req.body.confirm;
+  let email = req.params.id;
+  try {
+    email = jwt.verify(email, process.env.EMAIL_CRYPT).email_id;
+  } catch (error) {
+    return msg(res, 403, error);
+  }
+  console.log(email);
+  if (password != confirm)
+    return msg(res, 400, 'Bad request, Password Mismatch');
```
Comment (Anguandia) [inline src/controllers/passwordController.js:39]: translate

File: src/helpers/sendEmail.js [hunk 5]
```diff
@@ -33,7 +37,7 @@ function sendReset(email) {
     expiresIn: '1h',
   });
 
-  const url = `https://localhost:3000/api/v1/users/reset/${token}`;
+  const url = `${frontendUrl}/Reset?token=${token}`;
   const msg = `<h2>DOMINATORS</h2>
 <h4>Password Reset Link</h4>
 <div>
```
Comment (Anguandia) [inline src/helpers/sendEmail.js:38]: - This helper file has too much redundancy/repetition. When through with the feedback, consider refactoring this too
- Please revert to the original email credential variable names in the environment, i.e USER_EMAIL, USER_EMAIL_P. these others will cause bugs in existing features `PHANTOM_PASSWORD=
APP_MAIL= 
APP_MAIL_PASSWORD = 
USER_LOCALHOST_PASSWORD= 
USER_HOSTED_PASSWORD= `

**Review by Anguandia — CHANGES_REQUESTED** [review]
Good work @UmuhireJessie , please address the few concerns raised herein. Additionally, please replace the property name `Message` in ~/utils/responseHandler.js with `status`, most front-end code checks for the same property by the name of status

**Review by GilbertUp — APPROVED** [review]

File: src/controllers/passwordController.js (hunk 3 above)
Comment (UmuhireJessie) [reply src/controllers/passwordController.js:28]: Removed

File: src/helpers/sendEmail.js (hunk 5 above)
Comment (UmuhireJessie) [reply src/helpers/sendEmail.js:38]: Done!

**Review by fab-ryan — APPROVED** [review]

File: src/helpers/sendEmail.js (hunk 5 above)
Comment (Anguandia) [reply src/helpers/sendEmail.js:38]: Good!

File: .env.example (hunk 1 above)
Comment (Anguandia) [reply .env.example:17]: Please act on this one too! We'll merge for now, but please, fix it in the next available PR

**Review by Anguandia — APPROVED** [review]

**Pull request merged**